File: easy/3643_flip_square_submatrix_vertically/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<std::vector<int>>
  reverseSubmatrix(std::vector<std::vector<int>> &grid, int x, int y, int k) {
    size_t topRow = x;
    size_t bottomRow = x + k - 1;

    while (topRow < bottomRow) {
      for (size_t col = y; col < y + k; ++col)
        std::swap(grid[topRow][col], grid[bottomRow][col]);

      ++topRow;
      --bottomRow;
    }

    return grid;
  }
};

File: easy/3643_flip_square_submatrix_vertically/solution.py
from typing import List


class Solution:
    def reverseSubmatrix(
        self, grid: List[List[int]], x: int, y: int, k: int
    ) -> List[List[int]]:
        topRow = x
        bottomRow = x + k - 1

        while topRow < bottomRow:
            for col in range(y, y + k):
                grid[topRow][col], grid[bottomRow][col] = (
                    grid[bottomRow][col],
                    grid[topRow][col],
                )

            topRow += 1
            bottomRow -= 1

        return grid
